src/common/motion/position.ts

```diff
@@ -101,6 +101,7 @@ export class Position extends vscode.Position {
 
   private _nonWordCharRegex: RegExp;
   private _nonBigWordCharRegex: RegExp;
+  private _nonCamelCaseWordCharRegex: RegExp;
```

Comment:
I'm wondering if we should move away from having `position.ts` the dumping ground for all positioning related logic as it's a bit of a mess. 

As the changes you introduced to this file are specific to camel case, I wonder if we should create a new file under /plugins `camelcase.position.ts`?

What do you think?

Reply:
I'm not opposed to the idea in principle, but if I were to move I'd either have to duplicate or make public all the `get*Word*WithRegex` functions and maybe the `getAllPositions` family of functions since they're shared among all the word-based motions. Your call, I'm happy either way!